docs/content/docs/libs/state_processor_api.md

```diff
@@ -514,3 +512,120 @@ savepointWriter
+) WITH (
+  'connector' = 'savepoint',
+  'state.backend.type' = 'rocksdb',
+  'state.path' = '/root/dir/of/checkpoint-data/chk-1',
+  'operator.uid' = 'my-uid'
+);
+```
+
+### Connector options
+
+#### General options
+| Option             | Required | Default | Type                                          | Description |
+|--------------------|----------|---------|-----------------------------------------------|-------------|
+| connector          | required | (none)  | String                                        | Specify what connector to use, here should be 'state'. |
+| state.backend.type | required | (none)  | Enum Possible values: hashmap, rocksdb, forst | Defines the state backend which must be used for state reading. This must match with the value which was defined in Flink job which created the savepoint or checkpoint. |
+| state.path         | required | (none)  | String                                        | Defines the state path which must be used for state reading. All filesystem which are supported by Flink can be used here. |
+| operator.uid       | optional | (none)  | String                                        | Defines the operator UID which must be used for state reading (Can't be used together with UID hash). |
+| operator.uid.hash  | optional | (none)  | String                                        | Defines the operator UID hash which must be used for state reading (Can't be used together with UID). |
+
+#### Connector options for column ‘#’
+| Option                  | Required | Default | Type   | Description |
+|-------------------------|----------|---------|--------|-------------|
+| fields.#.state-name     | optional | (none)  | String | Overrides the state name which must be used for state reading. This can be useful when the state name contains characters which are not compliant with SQL column names. |
+| fields.#.state-type     | optional | (none)  | Enum Possible values: list, map, value | Defines the state type which must be used for state reading, including value, list and map. When It's not provided then it tries to be inferred from the SQL type (ARRAY=list, MAP=map, all others=value). |
+| fields.#.map-key-format | optional | (none)  | String | Defines the format class scheme for decoding map value key data (for ex. java.lang.Long). When It's not provided then it tries to be inferred from the SQL type (only primitive types supported). |
+| fields.#.value-format | optional | (none)  | String | Defines the format class scheme for decoding value data (for ex. java.lang.Long). When It's not provided then it tries to be inferred from the SQL type (only primitive types supported). |
+
+### Default Data Type Mapping
+
+The state SQL connector is inferring the data type for primitive types when `fields.#.value-format` and `fields.#.map-key-format`
+are not defined. The following table shows the `Flink SQL type` -> `Java type` default mapping. If the mapping is not calculated properly
```

Comment:
when can  "If the mapping is not calculated properly" occur?

Reply:
There is no 1to1 mapping between SQL->Java type.